src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,359 @@
+                                {
+                                    EnvironmentVariables = envVariables,
+                                    Parameters = parameters,
+                                    Platform = platform,
+                                    ExtensionFileName = extentionFileName,
+                                    PathToExtension = extentionFile
+                                });
+                            }
+                        }
+                    }
+                }
+                return new List<ExternalLanguage>(dic.Values);
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        private string GetCreateScript(ExternalLanguage language, Dictionary<string, object> parameters)
+        {
+            return GetLanguageModifyScript(language, language.Contents, parameters, ModifyType.Create);
+        }
+
+        private string GetUpdateScript(ExternalLanguage language, ExternalLanguageContent content, Dictionary<string, object> parameters, ContentModifyType contentModifyType)
+        {
+            return GetLanguageModifyScript(language, new List<ExternalLanguageContent> { content }, parameters, ModifyType.Alter, contentModifyType);
+        }
+
+        private string GetLanguageModifyScript(ExternalLanguage language, List<ExternalLanguageContent> contents, Dictionary<string, object> parameters, ModifyType modifyType, ContentModifyType contentModifyType = ContentModifyType.Add)
+        {
+            string contentScript = string.Empty;
+            for (int i = 0; i < contents.Count; i++)
+            {
+                var content = contents[i];
+                string seperator = contentScript == string.Empty ? "" : ",";
+                contentScript = $"{contentScript}{seperator}{GetLanguageContent(content, i, parameters)}";
+            }
+
+            string ownerScript = string.IsNullOrWhiteSpace(language.Owner) ? "" : $"AUTHORIZATION {language.Owner}";
+            string scriptAction = modifyType == ModifyType.Create ? CreateScript : AlterScript;
+            string contentAction = "FROM";
+            if (modifyType == ModifyType.Alter)
+            {
+                switch (contentModifyType)
+                {
+                    case ContentModifyType.Modify:
+                        contentAction = SetContentScript;
+                        break;
+                    case ContentModifyType.Add:
+                        contentAction = AddContentScript;
+                        break;
+                    case ContentModifyType.Remove:
+                        contentAction = RemoveContentScript;
+                        break;
+                }
+            }
+            return $@"
+{scriptAction} {language.Name}
+{ownerScript}
+{contentAction} {contentScript}
+";
+        }
+
+        private string AddStringParameter(string paramName, string prefix, string postfix, string paramValue, Dictionary<string, object> parameters)
```

Comment:
This seems like it should at the very least be a static function. Possibly even pull into a utils class but I'm not sure useful it is outside this context

Reply:
sure, I didn't need it other places. will move if I did. specially because I have problem with sql param for this specific tsql for now

---
In reply to: [374372900](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r374372900) [](ancestors = 374372900)